From 3d9b0d130a00dd32e6061ac708eaaaed69e35c3d Mon Sep 17 00:00:00 2001
From: Sebastian Lenartowicz <Sebastian.Lenartowicz@windriver.com>
Date: Wed, 6 Nov 2013 13:48:56 -0500
Subject: [PATCH] ARM: OMAP3: Fix USB PHY Power-On Issue on Beagleboard in
 Kernel 3.10

Changes in ehci-omap altered how power-on had to be handled for
peripherals. This fix ensures that the power for the USB PHY on the
Beagleboard is on at runtime.

Tested on a Beagleboard xM Rev. C2.

Signed-off-by: Sebastian Lenartowicz <Sebastian.Lenartowicz@windriver.com>
---
 arch/arm/mach-omap2/board-omap3beagle.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm/mach-omap2/board-omap3beagle.c b/arch/arm/mach-omap2/board-omap3beagle.c
index 04c1165..c385ec4 100644
--- a/arch/arm/mach-omap2/board-omap3beagle.c
+++ b/arch/arm/mach-omap2/board-omap3beagle.c
@@ -338,7 +338,6 @@ static int beagle_twl_gpio_setup(struct device *dev,
 	phy_data[0].vcc_gpio = gpio + TWL4030_GPIO_MAX;
 	phy_data[0].vcc_polarity = beagle_config.usb_pwr_level;
 
-	usbhs_init_phys(phy_data, ARRAY_SIZE(phy_data));
 	return 0;
 }
 
@@ -551,11 +550,12 @@ static void __init omap3_beagle_init(void)
 	omap_sdrc_init(mt46h32m32lf6_sdrc_params,
 				  mt46h32m32lf6_sdrc_params);
 
+	usbhs_init_phys(phy_data, ARRAY_SIZE(phy_data));
+	usbhs_init(&usbhs_bdata);
+	
 	usb_bind_phy("musb-hdrc.0.auto", 0, "twl4030_usb");
 	usb_musb_init(NULL);
 
-	usbhs_init(&usbhs_bdata);
-
 	board_nand_init(omap3beagle_nand_partitions,
 			ARRAY_SIZE(omap3beagle_nand_partitions), NAND_CS,
 			NAND_BUSWIDTH_16, NULL);
-- 
1.8.1.2

